src/lib/utils/semesters.ts

```diff
@@ -0,0 +1,43 @@
+export type Semester = number;
+
+// Semesters are encoded as $YEAR * 2 and +1 if its the fall term
```

Comment:
Why encode semesters like this?

Reply:
I needed a basic representation of a semester (i.e. not an object) so that I could put them in `Set`s to get the unique ones. This way ensures that sorting by semester gives the chronological order, since the fall semester is numerically bigger than the spring semester. 

With that said, you could also have gone with encoding VT 2023 as `20230` and HT 2023 as `20231`, or something like that. That approach would probably have been more readable, i guess.

Reply:
I also though about encoding them as strings, but if you encoded VT 2023 as `"VT 2023"` it would come after `"HT 2023"` by default.

Reply:
Fair enough. I added a suggestion for a more detailed comment explaining the reasoning.